### ISSUE TITLE
AGS 4 / new compiler: support dynamic arrays of normal structs

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#1818]
I've been wondering, if it's possible to support dynamic arrays of non-managed structs?
For example:

```
struct Cell
{
    int data;
};

Cell cells[];
```
Above results with error in the extended compiler:
> Can only have dynamic arrays of integer types, 'float', or managed structs. 'Cell' isn't any of this.

(old compiler does not error on declaration, but on the `new` operator instead)

Right now this is restricted in both old and new compilers. Although, theoretically, you should be able to store an element of any size in dynamic arrays. Perhaps there have been a problem in coding accessing these elements, or anything else I cannot think of right now?

**Comment by fernewelten** [comment]
At first sight I'd think that you're right, this _should_ work. I don't see any problems that could come up if this limitation were removed. We'd need to make sure, however, that the struct doesn't contain any dynamic entities, either directly or indirectly.

**Comment by fernewelten** [comment]
Yes, I could implement this, by eliminating the error message and adding a check to make sure that `Cell` doesn't contain any dynamic arrays or pointers. Do we want to have it? 

From the vantage point of the game coders, the code would be very similar to the code below that both the compilers can compile:
```
managed struct Cell
{
    int data;
};

Cell *cells[];
```

In the example above, each cell would be `null` and would need to be initialized individually with a ` cells[…]= new Cell;`. In Ivan's code, above, all the cells would be ready for use at the get-go.

**Comment by ivan-mogilko** [comment]
Note that compiler will still have to deny dynamic arrays of structs that have nested managed pointers, until #1923 is completed.

**Comment by ericoporto** [comment]
A note, that #1923 was finished and is working, so perhaps this could be done now with less checks.

**Comment by fernewelten** [comment]
Addressed in #2106

**Comment by ivan-mogilko** [comment]
Implemented by #2106.

Example of use:
```
struct Warrior
{
    int Strength;
    int Health;
};

function room_FirstLoad()
{
    Warrior Lemmings[] = new Warrior[10];
    player.Say("That one's strength is %d", Lemmings[1].Strength);
}
```
